shell/src/main/java/tachyon/shell/TfsShell.java

```diff
@@ -285,42 +242,59 @@ private int copyPath(File src, TachyonFileSystem tachyonClient, TachyonURI dstPa
             dstPath = dstPath.join(src.getName());
           }
         }
-      } catch (TachyonException e) {
-        throw new IOException(e.getMessage());
-      }
 
-      Closer closer = Closer.create();
-      try {
-        FileOutStream os =
-            closer.register(tachyonClient.getOutStream(dstPath, OutStreamOptions.defaults()));
-        FileInputStream in = closer.register(new FileInputStream(src));
-        FileChannel channel = closer.register(in.getChannel());
-        ByteBuffer buf = ByteBuffer.allocate(8 * Constants.MB);
-        while (channel.read(buf) != -1) {
-          buf.flip();
-          os.write(buf.array(), 0, buf.limit());
+        Closer closer = Closer.create();
+        FileOutStream os = null;
+        try {
+          os = closer.register(tachyonClient.getOutStream(dstPath, OutStreamOptions.defaults()));
+          FileInputStream in = closer.register(new FileInputStream(src));
+          FileChannel channel = closer.register(in.getChannel());
+          ByteBuffer buf = ByteBuffer.allocate(8 * Constants.MB);
+          while (channel.read(buf) != -1) {
+            buf.flip();
+            os.write(buf.array(), 0, buf.limit());
+          }
+        } catch (IOException e) {
+          // Close the out stream and delete the file, so we don't have an incomplete file lying
+          // around
+          if (os != null) {
+            os.cancel();
+            fd = tachyonClient.openIfExists(dstPath);
```

Comment:
Do you mean the `openIfExists` then `delete`? It seems like it's only used twice, and it's like 3 lines of code, so I'm not sure that's necessary yet.